api/src/main/java/org/apache/iceberg/ManageSnapshots.java

```diff
@@ -83,6 +83,19 @@ public interface ManageSnapshots extends PendingUpdate<Snapshot> {
    */
   ManageSnapshots cherrypick(long snapshotId);
 
+  /**
+   * Create a new branch. The branch will point to current snapshot if the current snapshot is not
+   * NULL. Otherwise, the branch will point to a newly created empty snapshot.
+   *
+   * @param name branch name
+   * @return this for method chaining
+   * @throws IllegalArgumentException if a branch with the given name already exists
```

Comment:
Would be nice to have a test for this.

Reply:
Let me do a follow-up.

Reply:
@rdblue I opened #8197 for a follow-up. And I am sorry for that indeed missed the checking.